### ISSUE TITLE
Implement model spin-up

### ISSUE THREAD

**Issue by dalonsoa** [ImperialCollegeLondon/virtual_rainforest#581]
From the proposal

- We need a mechanism to allow a model to run for many years while holding everything else constant. The principal use of this is to allow initial estimates of input data to converge to a biologically stable state before starting the ‘actual’ simulation from those stable conditions.
- There are two possible mechanisms for terminating the spin up process: run for a fixed number of years or run until a set of watched variables generate stable results to a given precision. It’s probably good to have both options and they could work together.
- In most cases, the spin up process can be thought of as a set of sequential runs with initial runs that we might want to converge to a steady state before we start the simulation, and then an actual simulation using outputs from those convergence runs. For example: 
  - Run models with all frozen apart from Hydrology for 10 years, save soil moisture. 
  - Run models with all frozen apart from Soil for 500 years, save carbon variables. 
  - Run models with all active for 100 years, using the saved outputs from the first two runs (the _actual_ simulation). 
- That _could_ be done manually using three runs and configs but it would be useful to have a configuration mechanism that: 
  - Specifies a set of spin up “epochs” with different model settings and spin up termination rules
  - Identifies which variables should pass from spin up into the main simulation.
  - In many cases, we will want to restart the ‘actual’ simulation with different scenarios after the burn in, so it might be cleaner to setup a burn in process that runs models and then updates the input data, ready for the actual simulation.
- The immediate use case here is the Hydrology model, but any model could conceivably want to use this process.

**Comment by dalonsoa** [comment]
@vgro , we will need example models to run and test if what we are implementing actually works, ideally the three scenarios described above of running hydrology only, soil only and finally the full thing.

**Comment by dalonsoa** [comment]
@vgro , do you have any update on this. I can give a start - probably next week - but proper testing will not be possible without some exemplar scenario to run.

**Comment by vgro** [comment]
@dalonsoa   I'm terribly sorry that I haven't replied to this, I never received an email about the issue and we haven't checked the issues systematically in a while. I have a few urgent tasks this week, I'll try and get something for you by the end of next week.

**Comment by vgro** [comment]
> @vgro , we will need example models to run and test if what we are implementing actually works, ideally the three scenarios described above of running hydrology only, soil only and finally the full thing.

Would this be a case for the static model?

**Comment by dalonsoa** [comment]
I think so. I still need to figure out how to implement it exactly, specially handling the inputs and outputs (such that the relevant outputs of the first run serve as inputs for the second, etc...).

I'm at a conferences the whole of this week, so no rush. If you can prepare the scenarios next week, I would schedule the start of the work on the sprint starting on the 27th of January.

**Comment by vgro** [comment]
> I think so. I still need to figure out how to implement it exactly, specially handling the inputs and outputs (such that the relevant outputs of the first run serve as inputs for the second, etc...).
> 
> I'm at a conferences the whole of this week, so no rush. If you can prepare the scenarios next week, I would schedule the start of the work on the sprint starting on the 27th of January.

We have started trying to set up a static model with 'hydrology only' but run into some issues in the first update, I'll keep you posted

**Comment by vgro** [comment]
@dalonsoa the static model is now running with the example data. To get started with this issue, I suggest using the example data set and setting everything but the hydrology to static (note that the plants have a separate config file).
As input you can repeat the example data, too.
Is this enough information to get started?

**Comment by dalonsoa** [comment]
Yes, I think so. I've schedule to work on this next week, so I'll keep you posted.
